plugins/modules/xen_orchestra_instance.py

```diff
@@ -0,0 +1,326 @@
+    label: This is a test from ansible
+    description: This is a test from ansible
+    boot_after_create: false
+
+- name: Start an existing virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: false
+    state: started
+
+- name: Stop an existing virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: false
+    state: stop
+
+- name: Restart an existing virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: false
+    state: stopped
+
+- name: Delete a virtual machine
+  community.general.xen_orchestra:
+    api_host: xen-orchestra.lab
+    user: user
+    password: passw0rd
+    validate_certs: false
+    state: absent
+'''
+
+import json
+import ssl
+import traceback
+from time import sleep
+
+from ansible_collections.community.general.plugins.module_utils.version import LooseVersion
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+
+# 3rd party imports
+try:
+    HAS_WEBSOCKET = True
+    WEBSOCKET_IMP_ERR = None
+    import websocket
+    from websocket import create_connection
+
+    if LooseVersion(websocket.__version__) < LooseVersion('1.0.0'):
+        raise ImportError
+except ImportError:
+    WEBSOCKET_IMP_ERR = traceback.format_exc()
+    HAS_WEBSOCKET = False
+
+OBJECT_NOT_FOUND = 1
+VM_STATE_ERROR = 13
+
+
+class XenOrchestra(object):
+    CALL_TIMEOUT = 100
+    '''Number of 1/10ths of a second to wait before method call times out.'''
+
+    def __init__(self, module):
+        # from config
+        self.counter = -1
+        self.con = None
+        self.module = module
+
+        self.create_connection(module.params['api_host'])
+        self.login(module.params['user'], module.params['password'])
+
+    @property
+    def pointer(self):
+        self.counter += 1
+        return self.counter
+
+    def create_connection(self, xoa_api_host):
+        validate_certs = self.module.params['validate_certs']
+        use_tls = self.module.params['use_tls']
+        proto = 'wss' if use_tls else 'ws'
+
+        sslopt = None if validate_certs else {'cert_reqs': ssl.CERT_NONE}
+        self.conn = create_connection(
+            '{0}://{1}/api/'.format(proto, xoa_api_host), sslopt=sslopt)
+
+    def call(self, method, params):
+        '''Calls a method on the XO server with the provided parameters.'''
+        id = self.pointer
+        self.conn.send(json.dumps({
+            'id': id,
+            'jsonrpc': '2.0',
+            'method': method,
+            'params': params
+        }))
+
+        waited = 0
+        while waited < self.CALL_TIMEOUT:
+            response = json.loads(self.conn.recv())
+            if 'id' in response and response['id'] == id:
```

Comment:
This can be simplified to:
```suggestion
            if response.get('id') == id:
```